src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6561,190 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
 
         public override IOperation? VisitInterpolatedStringHandlerCreation(IInterpolatedStringHandlerCreationOperation operation, int? captureIdForResult)
         {
-            return VisitNoneOperation(operation, captureIdForResult);
+            // We turn the interpolated string into a call to create the handler type, a series of append calls (potentially with branches, depending on the
+            // handler semantics), and then evaluate to the handler flow capture temp.
+
+            SpillEvalStack();
+            RegionBuilder resultRegion = CurrentRegionRequired;
+            var handlerCaptureId = captureIdForResult ?? GetNextCaptureId(resultRegion);
+            var previousHandlerId = _currentInterpolatedStringHandlerId;
+            _currentInterpolatedStringHandlerId = handlerCaptureId;
+
+            var constructorRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime);
+            EnterRegion(constructorRegion);
+
+            BasicBlockBuilder? resultBlock = null;
+            if (operation.HandlerCreationHasSuccessParameter || operation.HandlerAppendCallsReturnBool)
+            {
+                resultBlock = new BasicBlockBuilder(BasicBlockKind.Block);
+            }
+
+            // Any placeholders for arguments should have already been created, except for the out parameter if it exists.
+            int outParameterFlowCapture = -1;
+            IOperation? outParameterPlaceholder = null;
+
+            if (operation.HandlerCreationHasSuccessParameter)
+            {
+                // Only successful constructor binds will have a trailing parameter
+                Debug.Assert(operation.HandlerCreation is IObjectCreationOperation);
+                outParameterFlowCapture = GetNextCaptureId(constructorRegion);
+                var arguments = ((IObjectCreationOperation)operation.HandlerCreation).Arguments;
+                IArgumentOperation? outParameterArgument = null;
+
+                for (int i = arguments.Length - 1; i > 1; i--)
+                {
+                    if (arguments[i].ArgumentKind == ArgumentKind.Explicit)
```

Comment:
>arguments[i].ArgumentKind == ArgumentKind.Explicit [](http://example.com/codeflow?start=24&length=50)

We are looking for ```is IInterpolatedStringHandlerArgumentPlaceholderOperation { PlaceholderKind: InterpolatedStringArgumentPlaceholderKind.TrailingValidityArgument }``` and that is what we should be using as the condition rather than trying to infer this information from properties that do not have primiry relation to the fact.